model/src/order.rs

```diff
@@ -539,4 +491,29 @@ mod tests {
         assert_eq!(uid.to_string(), expected);
         assert_eq!(format!("{}", uid), expected);
     }
+
+    #[test]
+    fn order_signature_recovery() {
+        const PRIVATE_KEY: [u8; 32] =
+            hex!("0000000000000000000000000000000000000000000000000000000000000001");
+        let public_key = Account::Offline(PrivateKey::from_raw(PRIVATE_KEY).unwrap(), None);
```

Comment:
I think this can be `H160(PRIVATE_KEY)` so there is no need to go through the ethcontract types.

Reply:
nvm, I see that this has to calculate the public key from the private key. We could hardcode the pubic key or see if we can calulate it with secpk.

Reply:
Not so sure about this. Also, think we use the fact that it is an `Account`as well here.
<img width="431" alt="Screenshot 2021-03-15 at 13 37 35" src="https://user-images.githubusercontent.com/11778116/111154605-b0555080-8593-11eb-9935-648dd29462be.png">

Reply:
I could maybe use something like this though:

```
let context = &SECP256K1;
        let s: key::SecretKey = key::SecretKey::from_slice(context, &secret[..])?;
        let pub_key = key::PublicKey::from_secret_key(context, &s)?;
        let serialized = pub_key.serialize_vec(context, false);
        ```

Reply:
I think that makes sense and it's not too messy.

Reply:
Just need to compare `PublicKey` with `H160` now.